### PR TITLE
HPCC-16333 Document auth_feature attribute in DESDL

### PR DESCRIPTION
**Pull request by JamesDeFabia**
## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by JamesDeFabia** [comment]
@rpastrana Please review

**Comment by rpastrana** [comment]
@JamesDeFabia please squash

File: docs/DynamicESDL/DESDL-Mods/ESDLESPmethod.xml [hunk 1]
```diff
@@ -12,11 +12,13 @@
 
   <programlisting>
  ESPservice MyService
-
+
```
Comment (rpastrana) [inline docs/DynamicESDL/DESDL-Mods/ESDLESPmethod.xml]: this doesn't seem necessary

File: docs/DynamicESDL/DESDL-Mods/ESDLESPmethod.xml [hunk 2]
```diff
@@ -12,11 +12,13 @@
 
   <programlisting>
  ESPservice MyService
-
+ 
  {
    ESPmethod MyMethod1(MyMethod1Request, MyMethod1Response);
    ESPmethod 
+
```
Comment (rpastrana) [inline docs/DynamicESDL/DESDL-Mods/ESDLESPmethod.xml]: This doesn't seem necessary either

File: docs/DynamicESDL/DESDL-Mods/ESDLESPservice.xml [hunk 3]
```diff
@@ -13,6 +13,7 @@
   <para><emphasis role="bold">Example:</emphasis></para>
 
   <programlisting>ESPservice MyService
+  auth_feature("AllowMyService"),
```
Comment (rpastrana) [inline docs/DynamicESDL/DESDL-Mods/ESDLESPservice.xml]: ESPservice [auth_feature("AllowMyService")] MyService

File: docs/DynamicESDL/DESDL-Mods/ESDLauth_feature.xml [hunk 4]
```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!DOCTYPE book PUBLIC "-//OASIS//DTD DocBook XML V4.5//EN"
+"http://www.oasis-open.org/docbook/xml/4.5/docbookx.dtd">
+<sect2 id="ESDL_Attributes_auth_feature">
+  <title><emphasis role="bold">auth_feature</emphasis></title>
+
+  <para>The auth_feature attribute (valid only for an ESPService or ESPMethod)
+  allows you to specify means to authenticate user's permission to execute a
```
Comment (rpastrana) [inline docs/DynamicESDL/DESDL-Mods/ESDLauth_feature.xml]: ...means to verify user's...

File: docs/DynamicESDL/DESDL-Mods/ESDLauth_feature.xml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  method.</para>
+
+  <para>In order to enable this feature, your system must be configured to use
+  a form of security that supports feature level authentication, such as LDAP
+  security included in the Community edition of the platform. Once LDAP is
+  configured, you would add the tag name provided as the value for the
+  <emphasis role="bold">authFeature</emphasis> attribute to the feature level
+  authentication list in the Security section of ECL Watch. Then you would set
+  permissions for users and/or groups for each..</para>
```
Comment (rpastrana) [inline docs/DynamicESDL/DESDL-Mods/ESDLauth_feature.xml]: the end of the sentence seems chopped off, also there are two periods

File: docs/DynamicESDL/DESDL-Mods/ESDLauth_feature.xml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+  <para>If you are using a third-party Security Manager plugin, consult their
+  documentation for details on adding the tag name to their security
+  configuration.</para>
+
+  <para>The auth_feature is required for every method, but can be specified at
```
Comment (rpastrana) [inline docs/DynamicESDL/DESDL-Mods/ESDLauth_feature.xml]: The auth_feature attribute...

File: docs/DynamicESDL/DESDL-Mods/ESDLauth_feature.xml [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+  the ESPService level to apply to all methods within a service. You can
+  override for an individual method by setting the attribute at a method
+  level. </para>
+
+  <para>Setting auth_feature("NONE") means no authentication is needed.
+  Setting auth_feature("DEFERRED") defers the authentication to the business
+  logic in the ESP developer's method implementation logic</para>
+
+  <para></para>
```
Comment (rpastrana) [inline docs/DynamicESDL/DESDL-Mods/ESDLauth_feature.xml]: not sure if this is intentional

File: docs/DynamicESDL/DESDL-Mods/ESDLauth_feature.xml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+  <para><emphasis role="bold">Example:</emphasis></para>
+
+  <programlisting> ESPservice MyService [auth_feature("NONE")]
+
```
Comment (rpastrana) [inline docs/DynamicESDL/DESDL-Mods/ESDLauth_feature.xml]: this extra line can be distracting

**Review by rpastrana — COMMENTED** [review]
@JamesDeFabia a few comments.

**Comment by JamesDeFabia** [comment]
@rpastrana Edited and Squashed. Please have another look

**Comment by rpastrana** [comment]
@JamesDeFabia looks good.

**Comment by richardkchapman** [comment]
How does this relate to #9811 ?

**Pull request merged**